airflow-core/src/airflow/ui/src/i18n/README.md

```diff
@@ -17,8 +17,188 @@
+  - Reviewing the language aspects of translation-related Pull Requests (PRs).
+  - Resolving translation-related conflicts in PRs.
+  - Ensuring translation reflects current language usage and terminology.
+  - Resolving translation-related issues and discussions.
+
+### 5.2. Code owner
+
+- Code owners are responsible for the following, in their assigned supported translation, according to the procedures stated below:
+  - Reviewing the technical aspects of translation-related PRs (e.g., linting, formatting, etc.).
+  - Merging translation-related PRs approved by the translation owner.
+  - Resolving translation-related conflicts in PRs, when needed.
+  - Managing translation-related issues and discussions, when needed (e.g., closing issues).
+- Translations sponsors who function as code owners, are also responsible for ensuring that the translation owner is active and able to maintain the translation. If not, they should act according to section 6.4.
+
+### 5.3. Engaged translator
+
+- Engaged translators do not have any formal responsibilities, but they are encouraged to contribute to translations by:
+  - Suggesting improvements to existing translations.
+  - Reporting issues or inconsistencies in translations.
+  - Participating in discussions related to translations.
+  - Assisting translation owners with their tasks, when needed.
+  - Being 3rd party reviewers for translation-related PRs and conflicts, when needed.
+- They may be mentioned in a comment in the `.github/CODEOWNERS` file.
+- Suitable candidates for translation ownership may be suggested from engaged translators, upon their consent and approval by the procedure in section 6.1.
+
+## 6. Procedures
+
+### 6.1. Approval process of a translation owner
+
+- The designated code owner should post a thread in the dev. list to request for approval of the translation owner(s) for a supported translation.:
+    - Approvals of a translation owner who is also the code owner can be done by a lazy consensus.
+    - Approvals of translation owners who are non-committers require at least one binding vote of at least 1 PMC member, and no objections from committers/PMC.
+- Within the thread, the code owner should demonstrate that the translation owner is suitable for the role, according to the requirements in section 4.2.
+
+### 6.2. Approval process of a new translation
+
+The following steps outline the process for approving a new translation:
+> [!WARNING]
+> Before merging the PR, the committer should validate that files are in sync with the default language (English).
+
+- Creating a PR to add a new translation to the codebase ([see example](https://github.com/apache/airflow/pull/51258/files)), according to the standard and guidelines, which includes:
+    - The translation files in the `airflow-core/src/airflow/ui/src/i18n/locales/<LOCALE_CODE>` directory, where `<LOCALE_CODE>` is the code of the language (e.g., `fr` for French).
```

Comment:
Might be worth mentioning the acceptable standard for these entries (e.g. ISO 639-1 or 639-3). Perhaps it's worth transitioning the existing languages to the most recent standard, `639-3`, before it becomes too much work).